test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/Metadata/MetadataServiceTests.cs

```diff
@@ -125,15 +129,108 @@ public async Task GetTableInfoReturnsValidResults()
 
             await MetadataService.HandleGetTableRequest(metadataParmas, requestContext.Object);
 
-            DeleteTestTable(sqlConn);
+            DeleteTestTable(sqlConn, this.testTableSchema, this.testTableName);
 
             requestContext.VerifyAll();
         }
 
+        [Test]
+        public async Task VerifyGenerateServerContextualizationNotification()
+        {
+            this.testTableName += new Random().Next(1000000, 9999999).ToString();
+            this.testTableName2 += new Random().Next(0, 999999).ToString();
+
+            var connectionResult = LiveConnectionHelper.InitLiveConnectionInfo(null);
+            var sqlConn = ConnectionService.OpenSqlConnection(connectionResult.ConnectionInfo);
+
+            CreateTestTable(sqlConn, this.testTableSchema, this.testTableName);
+            CreateTestTable(sqlConn, this.testTableSchema, this.testTableName2);
+
+            var generateServerContextualizationParams = new GenerateServerContextualizationParams
+            {
+                OwnerUri = connectionResult.ConnectionInfo.OwnerUri
+            };
+
+            var service = new MetadataService();
+            var methodInfo = typeof(MetadataService).GetMethod("GenerateServerContextualization", BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
You shouldn't need to do this - the test assembly is already a friend to the servicelayer assembly so you should be able to call it directly. (we should avoid using reflection wherever possible)

https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Microsoft.SqlTools.ServiceLayer.csproj#L78

Reply:
It wasn't letting me access the method in the integration test, so I resorted to reflection. I tried to add the `InternalsVisisbleTo` attribute to the `GenerateServerContextualization` method too, but it's not recognizing `Microsoft.SqlTools.ServiceLayer.IntegrationTests` as an assembly name.

Reply:
Oh... I needed to switch the access specifier from `private` to `internal`